DashAI/front/src/components/experiments/PrepareDatasetStep.jsx

```diff
@@ -0,0 +1,398 @@
+      throw new Error(
+        "Ranges and indexes must match the format on the example above",
+      );
+    }
+    const ranges = value.split(",");
+    ranges.forEach((range) => {
+      const [min, max] = range.split("-");
+      if (!range.includes("-") && parseInt(range) <= total) {
+        numbersArray.push(parseInt(range));
+      } else if (
+        (!range.includes("-") && parseInt(range) > total) ||
+        parseInt(max) > total
+      ) {
+        throw new Error(
+          "The indexes should be minor than the total of columns or rows",
+        );
+      } else if (parseInt(min) > parseInt(max)) {
+        throw new Error(
+          "The second number of a range must be greater than the first",
+        );
+      } else {
+        for (let i = parseInt(min); i <= parseInt(max); i++) {
+          numbersArray.push(i);
+        }
+      }
+    });
+    return numbersArray;
+  };
+
+  const checkSplit = (training, validation, testing) => {
+    if (parseInt(training) + parseInt(validation) + parseInt(testing) === 100) {
+      return true;
+    } else {
+      return false;
+    }
+  };
+  const handleInputColumnsChange = (event) => {
+    const input = event.target.value.replace(/ /g, ""); // TODO: dont accept spaces between numbers
+    try {
+      const columnNumbers = parseRangeToIndex(input, totalColumns);
+      setParseInputColumnsError(false);
+      setInputColumns(columnNumbers);
+    } catch (error) {
+      setParseInputColumnsErrorText(error.message);
+      setParseInputColumnsError(true);
+    }
+  };
+  const handleOutputColumnsChange = (event) => {
+    const input = event.target.value.replace(/ /g, "");
+    try {
+      const columnIndex = parseRangeToIndex(input, totalColumns); // TODO: input and output columns should be less than total
+      setParseOutputColumnsError(false);
+      setOutputColumns(columnIndex);
+    } catch (error) {
+      setParseOutputColumnsErrorText(error.message);
+      setParseOutputColumnsError(true);
+    }
+  };
+  const handleRowsPreferenceChange = (event) => {
+    if (event.target.value === "manually") {
+      setRowsPartitionsPercentage(defaultPartitionsPercentage);
+    } else {
+      setRowsPartitionsIndex(defaultParitionsIndex);
+    }
+    setRowsPartitionsError(false);
+    setRowsPartitionsErrorText("");
+    setRowsPreference(event.target.value);
+  };
+
+  const handleRowsChange = (event) => {
+    const value = event.target.value;
+    const id = event.target.id; // TODO: check that the training, validation and testing rows dont overlap
+    if (rowsPreference === "manually") {
+      try {
+        const rowsIndex = parseRangeToIndex(value, totalRows);
+        setRowsPartitionsIndex(
+          id === "training"
+            ? { ...rowsPartitionsIndex, training: rowsIndex }
+            : id === "validation"
+            ? { ...rowsPartitionsIndex, validation: rowsIndex }
+            : { ...rowsPartitionsIndex, testing: rowsIndex },
```

Comment:
It's hard to read code like this! 
I don't think it's good to use 2 ternary operators in a row. 
I think there are better strategies like a switch case or something similar.


```suggestion
switch (id) {
  case "training":
    setRowsPartitionsIndex({ ...rowsPartitionsIndex, training: rowsIndex });
    break;
  case "validation":
    setRowsPartitionsIndex({ ...rowsPartitionsIndex, validation: rowsIndex });
    break;
  case "test":
    setRowsPartitionsIndex({ ...rowsPartitionsIndex, testing: rowsIndex });
    break;
}
```

This also applies to the following functions.